service/storage_service.cc

```diff
@@ -1126,6 +1126,53 @@ class topology_coordinator {
         }
     }
 
+    future<> enable_features(group0_guard guard, std::set<sstring> features_to_enable) {
+        if (!_topo_sm._topology.new_nodes.empty()) {
```

Comment:
Why is this here? This is not how the spec works. Nodes may still in request joining state while we enable features.

OK. I see that the patch description has:

"For now, the implementation relies on gossip shadow round check to protect from nodes without all features joining the cluster. In a followup, a new joining procedure will be implemented which involves the topology coordinator and lets it verify joining node's cluster features before the new node is added to group 0 and to the cluster."

So this patch series does not implement the spec, but then I do not know what it implements and cannot review it. Why can't we implement the spec directly?

Reply:
> So this patch series does not implement the spec, but then I do not know what it implements and cannot review it. Why can't we implement the spec directly?

The new joining procedure is still in the works. This PR represents what I managed to implement until now, and I sent it so that it can be reviewed quicker and maybe queued.